firebase-perf/src/main/java/com/google/firebase/perf/transport/TransportManager.java

```diff
@@ -448,7 +454,15 @@ private boolean isAllowedToDispatch(PerfMetric perfMetric) {
 
   @WorkerThread
   private void dispatchLog(PerfMetric perfMetric) {
-    logger.info("Logging %s", getLogcatMsg(perfMetric));
+
+    // Logs the metrics to logcat plus console URL for every trace metric.
+    if (perfMetric.hasTraceMetric()) {
```

Comment:
Can you abstract a separate utility method isTraceMetric(perfMetric) and use that method here? That way the logic of determining if the provided metric is a trace metric is abstracted.

Reply:
Good suggestion! However, I would like to do this in a separate PR. `hasTraceMetric()`, `hasNetworkRequestMetric()`, `hasGaugeMetric()` have been the ways to determine the type of the `PerfMetric` and are used [here](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-perf/src/main/java/com/google/firebase/perf/transport/TransportManager.java#L582-L596) and many other places in the codebase. Fixing all of these could be out of the scope of this PR.

Reply:
Sounds good. Thanks!